### PR TITLE
Make scan summary URL tenant-aware

### PR DESCRIPTION
**Pull request by armchairlinguist**
The findings URLs printed at the end of a Semgrep scan are not currently aware of the SEMGREP_URL/SEMGREP_APP_URL variable. For scans using this variable, the URLs were still pointing to semgrep.dev, which was not accurate. With this change, the variable is used rather than hardcoding to the standard URL.

### Testing

Manual test: 
`SEMGREP_APP_TOKEN={TOKEN} SEMGREP_APP_URL=https://staging.semgrep.dev/ pipenv run semgrep ci --include='*.py'`

Produced the following output, using the `SEMGREP_APP_URL` under View results:

```
CI scan completed successfully.
  Found 3 findings (0 blocking) from 21416 rules.
  Uploading scan results  
  Finalizing scan                                                                                                                                 
  View results in Semgrep Cloud Platform:
    https://staging.semgrep.dev/orgs/alexis-test/findings?repo=local_scan/semgrep&ref=armchairlinguist/tenant-aware-summary-links
    https://staging.semgrep.dev/orgs/alexis-test/supply-chain
  No blocking findings so exiting with code 0
```

(The repo and ref were recently added as well, since I started this PR.)

### CODE REVIEW

File: cli/tests/e2e-pro/test_ci.py [hunk 1]
```diff
@@ -895,7 +919,7 @@ def test_full_run(
     del scan_create_json["scan_metadata"]
     snapshot.assert_match(json.dumps(scan_create_json, indent=2), "meta.json")
 
-    findings_and_ignores_json = upload_results_mock.last_request.json()
+    findings_and_ignores_json = upload_results_mock().last_request.json()
```
Comment (brendongo) [inline cli/tests/e2e-pro/test_ci.py]:
```suggestion
    findings_and_ignores_json = upload_results_mock(env.get("SEMGREP_URL", "https://semgrep.dev/").last_request.json()
```

File: cli/tests/e2e-pro/test_ci.py [hunk 2]
```diff
@@ -1297,6 +1335,16 @@ def test_shallow_wrong_merge_base(
     subprocess.run(["git", "fetch", "origin", "--depth", "1", "bar:bar"])
     subprocess.run(["git", "checkout", "bar"], check=True, capture_output=True)
 
+    start_scan_mock = start_scan_mock_maker(
+        env.get("SEMGREP_URL", "https://semgrep.dev")
```
Comment (brendongo) [inline cli/tests/e2e-pro/test_ci.py]: Do these need to have env.get aka is SEMGREP_URL ever set on this test?

Comment (armchairlinguist) [reply cli/tests/e2e-pro/test_ci.py]: Not at this point, I'll remove it.

**Review by emjin — APPROVED** [review]
I'm not an expert on the test code conventions, but this looks good to me!

**Comment by armchairlinguist** [comment]
@emjin Thanks for the review! @brendongo and I worked on the tests together - Pad and Cooper pointed me to him as someone who would be more familiar with those tests. Totally okay if you'd still like other eyes on it though, I can request one of them as a reviewer also.

**Pull request merged**